### PR TITLE
Introduce pytest-splits

### PR DESCRIPTION
**Pull request by vincentmr**
### Before submitting

Please complete the following checklist when submitting a PR:

- [x] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [x] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [x] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Native MCM tests are very slow.

**Description of the Change:**
Split test jobs into a wheel building job and a test job. The test job is parametrized over `pytest-split`'s `group` so that several runners can run in parallel. The `all` tests are removed since LQ is now always installed alongside LK (at least in a no-bin version). 

**Benefits:**
Faster CI.

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-62128]

### CODE REVIEW

File: .github/workflows/python_lightning_kokkos_test_durations
Comment (AmintorDusko) [inline .github/workflows/python_lightning_kokkos_test_durations:1]: How are we using this file?

File: .github/workflows/python_lightning_qubit_test_durations
Comment (AmintorDusko) [inline .github/workflows/python_lightning_qubit_test_durations:1]: Same here. How are we using this?

**Review by AmintorDusko — COMMENTED** [review]
Looks good. I just want first to understand the role of these two files with test durations.

File: .github/workflows/python_lightning_kokkos_test_durations
Comment (vincentmr) [reply .github/workflows/python_lightning_kokkos_test_durations:1]: They are timings for every tests. They are used by `pytest-split` to split time evenly between groups. Doesn't seem to balance that well, but better than nothing.

File: .github/workflows/python_lightning_kokkos_test_durations [hunk 1]
```diff
@@ -0,0 +1,4523 @@
+{
```
Comment (mlxd) [inline .github/workflows/python_lightning_kokkos_test_durations:1]: Should this file be `name.json`, or something else?

File: .github/workflows/python_lightning_qubit_test_durations
Comment (mlxd) [inline .github/workflows/python_lightning_qubit_test_durations:1]: Should this also be `name.json` ?

File: .github/workflows/tests_lkcuda_python.yml [hunk 2]
```diff
@@ -105,7 +105,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-22.04]
-        pl_backend: ["lightning_kokkos", "all"]
+        pl_backend: ["lightning_kokkos"]
```
Comment (mlxd) [inline .github/workflows/tests_lkcuda_python.yml:108]: Should we be testing all here too, or is it tested elsewhere?

**Review by mlxd — COMMENTED** [review]
Thanks @vincentmr 
Just a few q's

File: .github/workflows/python_lightning_kokkos_test_durations (hunk 1 above)
Comment (vincentmr) [reply .github/workflows/python_lightning_kokkos_test_durations]: Yes, I'll rename with a `json` extension.

File: .github/workflows/python_lightning_qubit_test_durations
Comment (vincentmr) [reply .github/workflows/python_lightning_qubit_test_durations:1]: Yes, I'll rename with a `json` extension.

File: .github/workflows/tests_lkcuda_python.yml (hunk 2 above)
Comment (vincentmr) [reply .github/workflows/tests_lkcuda_python.yml]: Yes, we wanted to keep one instance of `"all"` testing. I would suggest keeping this one and testing only with `pl-device-test` as discussed in Slack.

**Review by AmintorDusko — APPROVED** [review]
Nice job! I'm sure this will help.

File: .github/workflows/tests_lkcpu_python.yml [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+  build_lightning_kokkos_wheels:
+    needs: [build_and_cache_Kokkos]
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        pl_backend: ["lightning_kokkos"]
+        exec_model: ${{ fromJson(needs.build_and_cache_Kokkos.outputs.exec_model) }}
+        kokkos_version: ${{ fromJson(needs.build_and_cache_Kokkos.outputs.kokkos_version) }}
+    timeout-minutes: 60
```
Comment (multiphaseCFD) [inline .github/workflows/tests_lkcpu_python.yml:51]: Is 60 minutes sufficient?

**Review by multiphaseCFD — APPROVED** [review]
Thanks @vincentmr ! Looks good to me. just a few minor question. Please feel free to merge it.

File: .github/workflows/tests_lkcpu_python.yml (hunk 3 above)
Comment (vincentmr) [reply .github/workflows/tests_lkcpu_python.yml:51]: Yes.

File: .github/workflows/tests_lgpumpi_python.yml [hunk 4]
```diff
@@ -128,7 +128,7 @@ jobs:
         run: |
           source /etc/profile.d/modules.sh && module use /opt/modules/ && module load ${{ matrix.mpilib }}/cuda-${{ matrix.cuda_version_maj }}.${{ matrix.cuda_version_min }}
           CMAKE_ARGS="-DCMAKE_C_COMPILER=mpicc -DCMAKE_CXX_COMPILER=mpicxx -DENABLE_MPI=ON -DCMAKE_CUDA_COMPILER=$(which nvcc) -DCMAKE_CUDA_ARCHITECTURES=${{ env.CI_CUDA_ARCH }} -DPython_EXECUTABLE=${{ steps.python_path.outputs.python }}" \
-          PL_BACKEND=lightning_gpu python -m pip install -e . --verbose
+          PL_BACKEND=lightning_gpu python -m pip install . --verbose
```
[the remark this answers is not in the document]
Comment (vincentmr) [reply .github/workflows/tests_lgpumpi_python.yml:131]: I (or we) do not understand why, but `-e` was causing issues in the wheel builds of #689 . Recent versions of `pip` and `-e` do not work with C++ backends either.  I'm thus thinking it is safer to use `-e` only when necessary.

[the remark this answers is not in the document]
Comment (AmintorDusko) [reply .github/workflows/tests_lgpumpi_python.yml:131]: I don't understand too. So it is we in this case.

**Pull request merged**